src/libstore/derivations.cc

```diff
@@ -103,31 +103,72 @@ StorePath writeDerivation(Store & store,
 }
 
 
-/* Read string `s' from stream `str'. */
-static void expect(std::istream & str, std::string_view s)
+/* Derivation parser local to this file.  Keep in unnabed namespace
+   to unblock aggressive inlining. */
+namespace {
+
+struct DerivationParserContext {
+    const char * ptr_;
+    const char * end_;
+
+    DerivationParserContext(const char * b, const char * e)
+      : ptr_(b) , end_(e)
+    {}
+    char peek(void) {
+        if (ptr_ == end_)
+            throw FormatError("unexpected EOF");
```

Comment:
```suggestion
            throw FormatError("Unexpected EOF");
```

Reply:
Initial version used uppercase in both places. One of the suggestions applied was to lower-case the first occurence (with assumption that rest of occurences will be lower-cased after the other material changes).

All entries are lower-case now.